heltour/tournament/views.py

```diff
@@ -1345,6 +1345,59 @@ def team(season):
+        
+        def season_performance(season):
+            season_score = 0
+            season_score_total = 0
+            season_perf = PerfRatingCalc()
+
+            games = defaultdict(list)
+            if season is None:
+                byes = {}
+            elif season.league.competitor_type == 'team':
+                pairings = TeamPlayerPairing.objects.filter(white=player) | TeamPlayerPairing.objects.filter(black=player)
+                for p in pairings.filter(team_pairing__round__season=season).order_by('team_pairing__round__number').nocache():
+                    games[p.team_pairing.round.number].append(p)
+                byes = {}
+            else:
+                pairings = LonePlayerPairing.objects.filter(white=player) | LonePlayerPairing.objects.filter(black=player)
+                for p in pairings.filter(round__season=season).order_by('round__number').nocache():
+                    games[p.round.number].append(p)
+                byes = {b.round.number: b for b in PlayerBye.objects.filter(round__season=season, player=player)}
+
+            for round_ in season.round_set.filter(publish_pairings=True).order_by('number'):
+                if round_.number in games:
+                    for p in games[round_.number]:
+                        if p.result == '':
+                            continue
+                        game_score = p.white_score() if p.white == player else p.black_score()
+                        if game_score is not None:
+                            season_score += game_score
+                            season_score_total += 1
+                        # Add pairing to performance calculation
+                        if p.game_played() and p.white is not None and p.black is not None:
+                            sp = SeasonPlayer.objects.filter(season=season, player=p.black if p.white == player else p.white).first()
+                            if sp is not None and sp.seed_rating is not None:
+                                opp_rating = sp.seed_rating
+                            else:
+                                opp_rating = p.black_rating_display() if p.white == player else p.white_rating_display(self.league)
+                            season_perf.add_game(game_score, opp_rating)
+                elif round_.number in byes:
+                    bye = byes[round_.number]
+                    season_score += bye.score()
+                    season_score_total += 1
+            return season_perf
+
+        def league_performance(seasons):
+            league_perf_obj = PerfRatingCalc()
+            for season in seasons:
+                league_perf_obj.merge(season_performance(season[0]))
+            return league_perf_obj
+
+        for leaguetype in other_season_leagues:
+            if leaguetype[0].tag == self.league.tag:
+                career = league_performance(leaguetype[1])
+                performances = [career.calculate(), career._score, career._game_count]
```

Comment:
Avoid accessing private (`_`-prefixed) members. In this case, you probably want the overall game count rather than only games eligible for performance rating. Those are the `season_score` and `season_score_total` variables in the season_performance function.

If you like, you can expand the PerfRatingCalc class to include this data. For example add an optional parameter,
`add_game(self, score, opponent_rating, include_in_perf=True):`
And then a couple extra fields (public, or accessible via a `@property`) for `score` and `score_total`.

Also, you probably meant to create a tuple (with `()` brackets) instead of a list (with `[]`).

Reply:
I wasn't really sure what you meant on this comment but I've copied the way it works for a single season instead now with:
        career_score = 0
        career_score_total = 0
        career_perf = PerfRatingCalc()